### PR TITLE
Better Configuration

### LINKED ISSUE #407: Allow configuration through ENV variables
This would make it easier to configure Obsidian when hosting it through docker, since no files will have to be modified manually.

### PR DESCRIPTION
**Pull request by Tides**
Replaces the old IServerConfiguration and uses the ConfigurationManager provided by Generic Host. This also makes it possible to now use environment variables to configure the server. As long as you're familiar with how people do it in asp net Core, the same would apply here. Another PR to generally improve and make this feel more .net like 🙏🏼 closes #407

### CODE REVIEW

File: Obsidian.ConsoleApp/config/server.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (Craftplacer) [inline Obsidian.ConsoleApp/config/server.json:1]: should this be committed?

**Review by Craftplacer — COMMENTED** [review]
Aside from half of the changes you've done not matching the title of the pull request, this look fine.

**Comment by Tides** [comment]
> Aside from half of the changes you've done not matching the title of the pull request, this look fine.

Lets just say they were all "necessary" changes and not me going on a minor side quest 😭 😭

File: Obsidian.ConsoleApp/config/server.json (hunk 1 above)
Comment (Tides) [reply Obsidian.ConsoleApp/config/server.json:1]: yessir

Comment (Naamloos) [reply Obsidian.ConsoleApp/config/server.json:1]: are these embedded resources that are copied when the server first launches?

**Comment by Naamloos** [comment]
looks good. Just curious about the logic of the pregenerated config files being there.

other than that, this closes #407 when merged.

File: Obsidian.ConsoleApp/config/server.json (hunk 1 above)
Comment (Tides) [reply Obsidian.ConsoleApp/config/server.json:1]: They're generated when you build or publish so it'll be included in artifacts and releases

**Review by Craftplacer — APPROVED** [review]

**Pull request merged**